managers/forms.py

```diff
@@ -12,27 +9,26 @@
 from users.models import CustomUser
 
 
-class ManagerSelectMultiple(Select2MultipleWidget):
-    def optgroups(self, name: str, value: List, attrs: Optional[Dict] = None) -> List:
-        self.choices.queryset = CustomUser.objects.exclude(user_type=CustomUser.UserType.EMPLOYEE.name).exclude(
-            pk=self.user_pk if hasattr(self, "user_pk") else None
-        )
-        return super().optgroups(name, value, attrs)
-
-
 class ProjectAdminForm(forms.ModelForm):
     def __init__(self, *args: Any, **kwargs: Any) -> None:
+        user_pk = kwargs.pop("user_pk") if "user_pk" in kwargs else None
         super().__init__(*args, **kwargs)
-        if hasattr(self, "user_pk"):
-            ManagerSelectMultiple.user_pk = self.user_pk
+
+        if "managers" in self.fields:
+            managers_to_choose = CustomUser.objects.exclude(user_type=CustomUser.UserType.EMPLOYEE.name)
+            if self.instance.pk:
+                self.fields["managers"].queryset = managers_to_choose
+            else:
```

Comment:
could you please explain `else`?

Reply:
Sure.
if instance.pk exist is mean that we user want to update project so we would like to show all available managers in current project  and all managers in db to choose. Furthermore, now this field is required because we must be sure that user of request will not delete all managers from project.
`else` mean that we are in `CreateProjectView` so as it said in issue we wanted to assigne user of request to managers and exclude him from choices, what is more, if we don't want to add any other managers I make this field optional.